freqtrade/data/converter/orderflow.py

```diff
@@ -0,0 +1,277 @@
+                df.loc[is_between, "trades"] = df.loc[is_between, "trades"].apply(
+                    lambda _: trades_grouped_df
+                )
+                # calculate orderflow for each candle
+                df.loc[is_between, "orderflow"] = df.loc[is_between, "orderflow"].apply(
+                    lambda _: trades_to_volumeprofile_with_total_delta_bid_ask(
+                        pd.DataFrame(trades_grouped_df), scale=config_orderflow["scale"]
+                    )
+                )
+                # calculate imbalances for each candle's orderflow
+                df.loc[is_between, "imbalances"] = df.loc[is_between, "orderflow"].apply(
+                    lambda x: trades_orderflow_to_imbalances(
+                        x,
+                        imbalance_ratio=config_orderflow["imbalance_ratio"],
+                        imbalance_volume=config_orderflow["imbalance_volume"],
+                    )
+                )
+
+                _stacked_imb = config_orderflow["stacked_imbalance_range"]
+                df.loc[is_between, "stacked_imbalances_bid"] = df.loc[
+                    is_between, "imbalances"
+                ].apply(lambda x: stacked_imbalance_bid(x, stacked_imbalance_range=_stacked_imb))
+                df.loc[is_between, "stacked_imbalances_ask"] = df.loc[
+                    is_between, "imbalances"
+                ].apply(lambda x: stacked_imbalance_ask(x, stacked_imbalance_range=_stacked_imb))
+
+                buy = df.loc[is_between, "bid"].apply(
+                    lambda _: np.where(
+                        trades_grouped_df["side"].str.contains("buy"),
+                        0,
+                        trades_grouped_df["amount"],
+                    )
+                )
+                sell = df.loc[is_between, "ask"].apply(
+                    lambda _: np.where(
+                        trades_grouped_df["side"].str.contains("sell"),
+                        0,
+                        trades_grouped_df["amount"],
+                    )
+                )
+                deltas_per_trade = sell - buy
+                min_delta = 0
+                max_delta = 0
+                delta = 0
+                for deltas in deltas_per_trade:
+                    for d in deltas:
+                        delta += d
+                        if delta > max_delta:
+                            max_delta = delta
+                        if delta < min_delta:
+                            min_delta = delta
+                df.loc[is_between, "max_delta"] = max_delta
+                df.loc[is_between, "min_delta"] = min_delta
+
+                df.loc[is_between, "bid"] = np.where(
+                    trades_grouped_df["side"].str.contains("buy"), 0, trades_grouped_df["amount"]
+                ).sum()
+                df.loc[is_between, "ask"] = np.where(
+                    trades_grouped_df["side"].str.contains("sell"), 0, trades_grouped_df["amount"]
+                ).sum()
```

Comment:
are you sure this condition is correct?
wouldn't you want the "ask" side of the OB to be the same as "sell" (and bid the same as buy)?

np.where's syntax is `np.where(condition, value_if_true, value_if_false)` ... so the "asks" line would use "0" in case of a sell side trade?

Reply:
So I totally understand where you are coming from logically and honestly I can't remember anymore why I wrote it that way. BUT I compared the results of it with what ATAS reports and this way it lines up. If I flip it around, it doesn't make sense anymore.

2 examples, first one with the current way it's done:
![image](https://github.com/freqtrade/freqtrade/assets/8218910/b3f529d1-cc8c-4303-af00-0fd6b078d95b)

second one with it flipped:
![image](https://github.com/freqtrade/freqtrade/assets/8218910/6965d630-b614-47cb-af2a-193b5418a313)


I'm still investigating this further, but maybe you got some sort of insight as well

Reply:
I may have figured it out, flipped ask/bid & buy/sell along the ways

Reply:
Looking at your comparison chart (the first one) - it looks rather great - having most numbers align.

any idea why the last 2 values (max_delta, total_trades) is so far off? 
just for 14:55, you've got a max delta of 141 ... while the left side has 706 - and similar for trades - with 10136 on the left, and ~4500 on the terminal side ...

Reply:
`max_delta` (& `min_delta`) does line up (and still do after the fix), it's just that I outputted it in the wrong order, so that probably has confused you. the "706" you referenced is the `volume`

`total_trades` on the other hand does not line up and as far as I have observed never has. I haven't figured out why so far. At first I thought they might count them twice, but this is not constant behavior

Reply:
hm interesting - so the assumption is that it's the ATAS report that's wrong ?

Reply:
Maybe. Or I haven't figured it out yet. So far it's just a simple line of code:
```py
                df.loc[is_between, "total_trades"] = len(trades_grouped_df)
```
so it's not like super complicated that I could mess up by accident. I will comb through the returned dataframe again and see if there is some column in there that acts as multiplier or something

Reply:
> I will comb through the returned dataframe again and see if there is some column in there that acts as multiplier or something

No, doesn't seem like it :thinking: